src/bpftrace.cpp

```diff
@@ -718,28 +718,59 @@ std::vector<std::unique_ptr<IPrintable>> BPFtrace::get_arg_values(const std::vec
     switch (arg.type.type)
     {
       case Type::integer:
-        switch (arg.type.size)
+        if (arg.type.IsSigned())
```

Comment:
The reason this fix works is b/c we're forcing a sign extension here, right? If so, can you add that information to the commit message?

Reply:
@danobi I've removed the confusing cast hack thing and just have Printables handle the actual printing, should be easier to make sense of now.